src/libsystemd/sd-netlink/netlink-genl.c

```diff
@@ -261,9 +277,10 @@ static int genl_family_get_by_name_internal(
         if (r < 0)
                 return r;
 
-        r = sd_netlink_call(nl, req, 0, &reply);
-        if (r < 0)
-                return r;
+        if (sd_netlink_call(nl, req, 0, &reply) < 0) {
+                (void) genl_family_new_unsupported(nl, name, type_system, ret);
+                return -EOPNOTSUPP;
```

Comment:
So this returns a valid object now on failure? is that necessary? i mean, our coding style dictates taht our functions do not touch return params on failure...

If this really should be kept, then there should be a comment somewhere highlighting this anomaly. But quite frnakly i'd look for a different way to solve this.

i.e. maybe return 0 or 1 depending if the family is supported or not. i.e. return something >= 0 (i.e. sucess) in both cases.

Reply:
Ouch. Fixed.